src/matlab2tikz.m

```diff
@@ -4455,14 +4489,19 @@ function warnFacetedInterp(m2t)
     pgfplotsColormap = sprintf('colormap={%s}{[1%s] %s}',name, unit, join(m2t, colSpecs, '; '));
 end
 % ==============================================================================
-function fontStyle = getFontStyle(m2t, handle)
+function [m2t, fontStyle] = getFontStyle(m2t, handle)
     fontStyle = '';
     if strcmpi(get(handle, 'FontWeight'),'Bold')
         fontStyle = sprintf('%s\\bfseries',fontStyle);
     end
     if strcmpi(get(handle, 'FontAngle'), 'Italic')
         fontStyle = sprintf('%s\\itshape',fontStyle);
     end
+    if ~all(get(handle, 'Color')==0)
+        color = get(handle, 'Color');
+        [m2t, col] = getColor(m2t, handle, color, 'patch');
+        fontStyle = sprintf('%s\\color{%s}', fontStyle, col);
+    end
```

Comment:
After further exploration, I decided the right thing to do is to move the coloring for text, axis labels, and titles all into here.